pkg/storage/manifest/storage.go

```diff
@@ -1,22 +1,40 @@
 package manifest
 
 import (
-	"github.com/weaveworks/ignite/pkg/apis/ignite/scheme"
-	"github.com/weaveworks/ignite/pkg/constants"
+	"github.com/weaveworks/ignite/pkg/serializer"
 	"github.com/weaveworks/ignite/pkg/storage"
 	"github.com/weaveworks/ignite/pkg/storage/sync"
 	"github.com/weaveworks/ignite/pkg/storage/watch"
 )
 
-func NewManifestStorage(dataDir string) (*ManifestStorage, error) {
-	ws, err := watch.NewGenericWatchStorage(storage.NewGenericStorage(storage.NewGenericMappedRawStorage(dataDir), scheme.Serializer))
+// NewManifestStorage constructs a new storage that watches unstructured manifests in the specified directory,
+// decodable using the given serializer.
+func NewManifestStorage(manifestDir string, ser serializer.Serializer) (*ManifestStorage, error) {
+	ws, err := watch.NewGenericWatchStorage(storage.NewGenericStorage(storage.NewGenericMappedRawStorage(manifestDir), ser))
+	if err != nil {
+		return nil, err
+	}
+
+	ss := sync.NewSyncStorage(ws)
+
+	return &ManifestStorage{
+		Storage: ss,
+	}, nil
+}
+
+// NewManifestStorage constructs a new storage that watches unstructured manifests in the specified directory,
+// decodable using the given serializer. However, all changes in the manifest directory, are also propagated to
+// the structured data directory that's backed by the default storage implementation. Writes to this storage are
+// propagated to both the manifest directory, and the data directory.
+func NewTwoWayManifestStorage(manifestDir, dataDir string, ser serializer.Serializer) (*ManifestStorage, error) {
```

Comment:
`TwoWay` might be a bit confusing as this doesn't propagate data directory changes to the manifest directory (like the name implies). Is `DataManifestStorage` better?

Reply:
It does propagate all the manifest that it knows about from before, so I think this is ok for now at least. We can talk about naming later.